yaml-tests/src/main/java/com/apple/foundationdb/relational/yamltests/command/QueryExecutor.java

```diff
@@ -188,6 +197,22 @@ private Continuation executeQuery(@Nonnull YamlConnection connection, @Nonnull Q
         return continuationAfter;
     }
 
+    @Nullable
+    private Continuation executeWithSetup(final @Nonnull YamlConnection connection,
+                                          SQLFunction<YamlConnection, Continuation> execute) throws SQLException, RelationalException {
+        if (!setup.isEmpty()) {
+            return connection.executeTransactionally(singleConnection -> {
+                for (var setupStatement : setup) {
+                    final RelationalStatement statement = singleConnection.createStatement();
+                    statement.execute(setupStatement);
```

Comment:
Hmm, while I see this to be working well for creating temporary functions, I am a unclear on the scope of SQL statements that can be used in "setup". For instance, `UPDATE .... RETURNING ...` won't work here as it requires slurping the result set. Now that I think of it, DQL works fine and so does DML except the case I mentioned above.

Reply:
I think we should either, 
1) restrict the scope of statements here (maybe to only CREATE TEMPORARY FUNCTION ....) by some sort of patern matching, or we can error if the `execute` returns a `ResultSet` (barring DQL completely). 
2) or, we could softly specify in documentation about its specific usages and cautions.   
 
 Open to your thoughts on this!

Reply:
Yeah, that's a good callout.
We currently commit at the end of the query (maybe that's a mistake), and so having other DDL would cause problems. I forbid it for now, and put a comment there that this might not stay if we can figure out how we want to use it in other situations.